### PR TITLE
Update file_integrity logging

### PR DESCRIPTION
**Pull request by andrewkroh**
- Use nanoseconds for logging time.Duration

- Add Digest type implementing TextMarshaler

  This adds `type Digest []byte` for representing the result of a hash function.

  This new type implements the Stringer and TextMarshaler interfaces so that the digest value is represented as a hexadecimal string.

- Update file_integrity metricset logging

### CODE REVIEW

File: auditbeat/module/file_integrity/action.go [hunk 1]
```diff
@@ -83,6 +83,9 @@ func (action Action) String() string {
 	return strings.Join(list, "|")
 }
 
+// MarshalText marshals the Action to a string.
```
Comment (ruflin) [inline auditbeat/module/file_integrity/action.go]: Minor details but it marshals it to a string and converts it then to a byte array :-)

File: auditbeat/module/file_integrity/event.go [hunk 2]
```diff
@@ -73,15 +78,27 @@ var typeNames = map[Type]string{
...
+type Digest []byte
+
+// String returns the digest value in lower-case hexadecimal form.
+func (d Digest) String() string {
+	return hex.EncodeToString(d)
+}
+
+// MarshalText encodes the digest to a hexadecimal string. This implements the
+// encoding.TextMarshaler interface.
+func (d Digest) MarshalText() ([]byte, error) { return []byte(d.String()), nil }
```
Comment (ruflin) [inline auditbeat/module/file_integrity/event.go]: See above.

**Comment by andrewkroh** [comment]
jenkins, retest it

**Comment by andrewkroh** [comment]
Rebased to fix merge conflicts and squashed to 3 self-contained commits.

**Comment by andrewkroh** [comment]
jenkins, test it

File: auditbeat/module/file_integrity/scanner.go [hunk 3]
```diff
@@ -70,34 +67,34 @@ func (s *scanner) Start(done <-chan struct{}) (<-chan Event, error) {
 
 // scan iterates over the configured paths and generates events for each file.
 func (s *scanner) scan() {
-	if logp.IsDebug(metricsetName) {
-		debugf("%v File system scanner is starting for paths [%v].",
-			s.logID, strings.Join(s.config.Paths, ", "))
-		defer debugf("%v File system scanner is stopping.", s.logID)
-	}
+	s.log.Debugw("File system scanner is starting", "file_path", s.config.Paths)
+	defer s.log.Debug("File system scanner is stopping")
```
Comment (ph) [inline auditbeat/module/file_integrity/scanner.go]: +1 to remove the conditionals check.

**Pull request merged**